### PR TITLE
Implements few SidebarClass functions.

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

File: src/game/engine/sidebar.cpp [hunk 1]
```diff
@@ -123,19 +119,15 @@ SidebarClass::StripClass::StripClass(InitClass const &init) :
 
 void SidebarClass::StripClass::One_Time(int column)
 {
-    // TODO, needs HouseClass, TabClass, HelpClass.
-#ifdef GAME_DLL
-    void (*func)(const StripClass *, int) = reinterpret_cast<void (*)(const StripClass *, int)>(0x0054DD54);
-    func(this, column);
-#else
+    static char *SpecialWeaponFile[SPECIAL_COUNT] = { "sonr", "atom", "warp", "pbmb", "pinf", "cam", "infx", "gpss" };
```
Comment (OmniBlade) [inline src/game/engine/sidebar.cpp]: statics within functions should be named `_static_var_name`

File: src/game/engine/sidebar.cpp [hunk 2]
```diff
@@ -352,15 +366,33 @@ BOOL SidebarClass::StripClass::Factory_Link(int factory_id, RTTIType type, int i
...
-    return func(this, unk1);
-#else
-    return 0;
-#endif
+    bool abandon = false;
+    bool bool1 = true;
+
+    for (int i = 0; i < m_CameoCount; ++i) {
+        if (m_Entries[i].Factory == factory) {
+            FactoryClass *fptr = g_Factories.Raw_Ptr(factory);
```
Comment (OmniBlade) [inline src/game/engine/sidebar.cpp]: Should be `&g_Factories[factory];`.

File: src/game/engine/sidebar.cpp [hunk 3]
```diff
@@ -587,11 +636,6 @@ void SidebarClass::Refresh_Cells(cell_t cellnum, const int16_t *list)
 
 void SidebarClass::Reload_Sidebar()
 {
-// TODO Needs HouseClass
-#ifdef GAME_DLL
-    void (*func)(const SidebarClass *) = reinterpret_cast<void (*)(const SidebarClass *)>(0x0054D340);
-    func(this);
-#elif 0
     // in order of the houses.
     static char sidebarnames[HOUSES_COUNT][12] = { "side?na.shp",
```
Comment (OmniBlade) [inline src/game/engine/sidebar.cpp]: Follow style for static variables in a function, `_sidebar_names`.

File: src/game/engine/sidebar.cpp [hunk 4]
```diff
@@ -604,26 +648,25 @@ void SidebarClass::Reload_Sidebar()
         "side?na.shp",
         "side?us.shp" };
 
-    int side_index;
+    int side_index = 0;
```
Comment (OmniBlade) [inline src/game/engine/sidebar.cpp]: `int side_index = g_PlayerPtr != nullptr ? g_PlayerPtr->Acts_Like() : 0;`

**Review by OmniBlade — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**